### PR TITLE
Configure threshold with the same values as rules severity

### LINKED ISSUE #624: consistent use of severity values for rules and threshold
Tested with Robocop 2.1.0:

## Configuring rules
severity values **I/W/E** and **i/w/e** are accepted, e.g.
`-c line-too-long:severity:e`

## Configuring threshold:
only severity values **I/W/E** are accepted.
The setting of e.g.
`--threshold e`
is just ignored.

I would suggest that for severity and threshold the same data set is accepted.

### PR DESCRIPTION
**Pull request by bhirsz**
Fixes #624

### CODE REVIEW

File: robocop/rules.py [hunk 1]
```diff
@@ -74,7 +74,10 @@ def parser(cls, value: Union[str, "RuleSeverity"]) -> "RuleSeverity":
             "i": cls.INFO,
         }.get(str(value).lower(), None)
         if severity is None:
-            raise ValueError(f"Chose one of: {', '.join(sev.value for sev in cls)}") from None
+            severity_values = ", ".join(sev.value for sev in cls)
+            raise robocop.exceptions.InvalidArgumentError(
+                f"Invalid severity value '{value}'. Chose one of: {severity_values}."
```
Comment (mnojek) [inline robocop/rules.py:79]:
```suggestion
                f"Invalid severity value '{value}'. Choose one from: {severity_values}."
```

File: tests/e2e/test_e2e.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
 """ General E2E tests to catch any general issue in robocop """
```
Comment (mnojek) [inline tests/e2e/test_e2e.py:1]:
```suggestion
""" General E2E tests to catch any general issue in Robocop """
```

File: tests/e2e/test_e2e.py [hunk 3]
```diff
@@ -171,11 +179,19 @@ def test_use_nested_argument_file(self, test_data_dir):
...
-    def test_set_rule_threshold(self, robocop_instance, test_data_dir):
-        should_run_with_config(robocop_instance, f"--threshold E {test_data_dir}")
+    @pytest.mark.parametrize("threshold", ["i", "I", "e", "error", "W", "WARNING"])
+    def test_set_rule_threshold(self, threshold, robocop_instance, test_data_dir):
+        with mock.patch.object(sys, "argv", f"robocop --threshold {threshold}".split()):
+            config = Config(from_cli=True)
 
     def test_set_rule_invalid_threshold(self, robocop_instance, test_data_dir):
-        should_run_with_config(robocop_instance, f"--threshold 3 {test_data_dir}")
+        error = "Invalid configuration for Robocop:\nInvalid severity value '3'. Chose one of: I, W, E."
```
Comment (mnojek) [inline tests/e2e/test_e2e.py:188]:
```suggestion
        error = "Invalid configuration for Robocop:\nInvalid severity value '3'. Choose one from: I, W, E."
```

**Comment by bhirsz** [comment]
Thanks for the failing tests I've discovered why it was not consistent before - during refactor of Rule/RuleParam class we added generic handler for invalid params (param raise ValueError, RuleParam catches is and reraise it as InvalidParam exception). This handling was missing for threshold - that's why threshold defaulted to Info if there was invalid input. There was TODO to fix that but we decided to remove the TODO since we forgot why it was there in first place :P (note to myself: never put empty TODO).

**Review by mnojek — DISMISSED** [review]

**Review by mnojek — APPROVED** [review]

**Pull request merged**